### ISSUE TITLE
Throw an exception if a class name has a leading \

### ISSUE THREAD

**Issue by mnapoli** [mnapoli/PHP-DI#243]
When a class name is prefixed with `\` PHP-DI doesn't show any warning:

```php
return array(
    '\Foo\Bar' => DI\object(),
);
```

This will not work as FQN must not contain a leading `\`. PHP-DI should throw an exception.

Related to #242, might need a "validation pass"? Enabled only for dev environment?

**Comment by juliangut** [comment]
Wouldn't it be enough to test for `class_exists` and `interface_exists` if string begins with slash and if it does then throw the exception?

**Comment by mnapoli** [comment]
Yes the problem is "when" to do that? We can't do that everytime the definition array is read, that would mean iterating through all keys, it's not efficient.

I'm not sure how it can be done except by adding some sort of "validation" process. E.g. `$container->validate()` or something like that that users/frameworks could call in `dev` mode.
